cmd/usage/usage.go

```diff
@@ -55,93 +56,145 @@ var getCmd = &cobra.Command{
-		case "csv":
-			if err := outputCSV(usageData, outputFile); err != nil {
-				logrus.Fatalf("Error outputting CSV: %v", err)
-			}
-		case "json":
-			if err := outputJSON(usageData, outputFile); err != nil {
-				logrus.Fatalf("Error outputting JSON: %v", err)
-			}
-		default:
-			logrus.Warnf("Unsupported format: %s. Defaulting to CSV.", outputFormat)
-			if err := outputCSV(usageData, outputFile); err != nil {
-				logrus.Fatalf("Error outputting CSV: %v", err)
+	for _, cluster := range clusterData {
+		clusterName := cluster.Name
+		clusterUUIDMap[clusterName] = cluster.Id
+	}
+
+	var selectedUUIDs []string
+	if len(clusters) > 0 {
+		selectedUUIDs = make([]string, 0, len(clusters))
+		for _, clusterName := range clusters {
+			if clusterID, ok := clusterUUIDMap[clusterName]; ok {
+				selectedUUIDs = append(selectedUUIDs, clusterID)
+			} else {
+				logrus.Fatalf("Cluster name '%s' not found within the specified time range.\n", clusterName)
 			}
 		}
-	},
+	}
+
+	return selectedUUIDs
+}
+
+func validateAndFormatDates(startDate, endDate string) (formattedStartDate, formattedEndDate string, err error) {
+	// Validate start date and end date
+	startDateTime, err := parseAndFormatDate(startDate)
+	if err != nil {
+		return "", "", fmt.Errorf("invalid start date format: %v", err)
+	}
+
+	endDateTime, err := parseAndFormatDate(endDate)
+	if err != nil {
+		return "", "", fmt.Errorf("invalid end date format: %v", err)
+	}
+
+	if startDateTime.After(endDateTime) {
+		return "", "", fmt.Errorf("start date must be before end date")
+	}
+
+	// Format dates
+	formattedStartDate = startDateTime.Format("2006-01-02T15:04:05.000Z")
+	formattedEndDate = endDateTime.Format("2006-01-02T15:04:05.000Z")
+
+	return formattedStartDate, formattedEndDate, nil
+}
+
+func CheckOutputFile(outputFile, outputFormat string, startDateTime, endDateTime string) (combinedFilename, fileExtension string, err error) {
+	formattedStartDate, err := time.Parse("20060102T150405", startDateTime) // Format as YYYYMMDDHHmmSS
+	if err != nil {
+		logrus.Fatalf("Error: %v\n", err)
+	}
+	formattedEndDate, err := time.Parse("20060102T150405", endDateTime) // Format as YYYYMMDDHHmmSS
+	if err != nil {
+		logrus.Fatalf("Error: %v\n", err)
+	}
+
+	// Assigning default value to filename if not specified
+	outputFileFormat := "usage_%s_%s"
+	if outputFile == "" {
+		outputFile = fmt.Sprintf(outputFileFormat, formattedStartDate, formattedEndDate)
+	}
+
+	if outputFormat == "" {
+		outputFormat = "csv"
+	}
+	outputFormat = strings.ToLower(outputFormat)
+	combinedFilename = GetNormalizedFilePathForFormat(outputFile, outputFormat)
+
+	fileExtension = strings.ToLower(strings.TrimPrefix(filepath.Ext(combinedFilename), "."))
+
+	if outputFormat != "" && outputFormat != fileExtension {
+		logrus.Warnf("Mismatch between output file extension and output format. Using output format: %s", outputFormat)
+	}
+
+	// Check if the file already exists
+	if _, err := os.Stat(combinedFilename); err == nil {
```

Comment:
I think there should be a way to override/overwrite the file if it exists - perhaps the --force or --overwrite option - @cdavid ?

Reply:
Yes that sounds good.
I can add  ` -f, --force [OPTIONAL] Overwrite the output file if it exists`.
cc: @cdavid

Reply:
I am ok with both of them, I do think `overwrite` make more sense.

Reply:
We have something similar for cert download.
`downloadCertificate.Flags().BoolP("force", "f", false, "[OPTIONAL] Overwrite the output file if it exists")`.
I can use force to be consistent.

Reply:
ok, let's do that then

Reply:
Done, please TAL.